sympy/polys/matrices/tests/test_lll.py

```diff
@@ -0,0 +1,124 @@
+def test_lll():
+    from sympy.core.numbers import Rational
+    from sympy.polys.domains import ZZ
+    from sympy.polys.matrices.ddm import DDM
+    from sympy.polys.matrices.lll import ddm_lll
+    x = DDM(
+        [
+            [ZZ(1), ZZ(0), ZZ(0), ZZ(0), ZZ(-20160)],
+            [ZZ(0), ZZ(1), ZZ(0), ZZ(0), ZZ(33768)],
+            [ZZ(0), ZZ(0), ZZ(1), ZZ(0), ZZ(39578)],
+            [ZZ(0), ZZ(0), ZZ(0), ZZ(1), ZZ(47757)]
+        ],
+        (4, 5,),
+        ZZ
+    )
+    assert ddm_lll(x, delta=Rational(5, 6)) == DDM(
+        [
+            [ZZ(10), ZZ(-3), ZZ(-2), ZZ(8), ZZ(-4)],
+            [ZZ(3), ZZ(-9), ZZ(8), ZZ(1), ZZ(-11)],
+            [ZZ(-3), ZZ(13), ZZ(-9), ZZ(-3), ZZ(-9)],
+            [ZZ(-12), ZZ(-7), ZZ(-11), ZZ(9), ZZ(-1)]
+        ],
+        (4, 5),
+        ZZ
+    )
+
+
+def test_lll_linear_dependent():
+    from sympy.core.numbers import Rational
+    from sympy.polys.domains import ZZ
+    from sympy.polys.matrices.ddm import DDM
+    from sympy.polys.matrices.lll import ddm_lll
+    x = DDM(
+        [
+            [ZZ(0), ZZ(-1), ZZ(-2), ZZ(-3)],
+            [ZZ(1), ZZ(0), ZZ(-1), ZZ(-2)],
+            [ZZ(2), ZZ(1), ZZ(0), ZZ(-1)],
+            [ZZ(3), ZZ(2), ZZ(1), ZZ(0)]
+        ],
+        (4, 4),
+        ZZ
+    )
+    try:
+        ddm_lll(x, delta=Rational(5, 6))
+    except ZeroDivisionError:
+        pass
+
+
+def test_lll_ddm():
+    from sympy.core.numbers import Rational
+    from sympy.polys.domains import ZZ
+    from sympy.polys.matrices.ddm import DDM
+    x = DDM(
```

Comment:
Rather than duplicating the test examples in different formats it would be better to just list the examples once like:
```python
_lll_inputs_ouputs = [
   # input1
   # output1
   # input2
   # output2
]

def test_lll():
    for input, output in _lll_inputs_outputs:
        # test DomainMatrix
        assert input.lll() == output
        # test DDM
        assert input.rep.to_dense().lll == output.rep.to_dense()
        # test SDM
        assert input.rep.to_sparse().lll == output.rep.to_sparse()
```